plugin/restore/mysql/mysql.go

```diff
@@ -69,32 +76,37 @@ func (r *Restore) RestorePITR(ctx context.Context, fullBackup *bufio.Scanner, bi
 }
 
 // SwapPITRDatabase renames the pitr database to the target, and the original to the old database
-func (r *Restore) SwapPITRDatabase(ctx context.Context, database string, timestamp int64) error {
+// It returns the pitr and old database names after swap.
+// It performs the step 2 of the restore process.
+// TODO(dragonly): handle the case that the original database does not exist
+func (r *Restore) SwapPITRDatabase(ctx context.Context, database string, suffixTs int64) (pitrDatabaseName string, pitrOldDatabase string, err error) {
+	pitrDatabaseName = getPITRDatabaseName(database, suffixTs)
+	pitrOldDatabase = getPITROldDatabaseName(database, suffixTs)
+
 	db, err := r.driver.GetDbConnection(ctx, "")
 	if err != nil {
-		return err
+		return
 	}
 	txn, err := db.BeginTx(ctx, nil)
 	if err != nil {
-		return err
+		return
 	}
 	defer txn.Rollback()
 
-	pitrOldDatabase := getSafeName(database, "old")
-	pitrDatabaseName := getPITRDatabaseName(database, timestamp)
-
-	if _, err := txn.ExecContext(ctx, fmt.Sprintf("CREATE DATABASE `%s`", pitrOldDatabase)); err != nil {
-		return err
-	}
-
-	// TODO(dragonly): handle the case that the original database does not exist
 	tables, err := mysql.GetTables(txn, database)
 	if err != nil {
-		return fmt.Errorf("failed to get tables of database %q, error[%w]", database, err)
+		err = fmt.Errorf("failed to get tables of database %q, error[%w]", database, err)
+		return
 	}
 	tablesPITR, err := mysql.GetTables(txn, pitrDatabaseName)
 	if err != nil {
-		return fmt.Errorf("failed to get tables of database %q, error[%w]", pitrDatabaseName, err)
+		err = fmt.Errorf("failed to get tables of database %q, error[%w]", pitrDatabaseName, err)
+		return
+	}
+
+	// TODO(dragonly): CREATE DATABASE is non-transactional, we should drop it if anything fails in the whole process later.
```

Comment:
Do we also need a pre-condition check that tables and tablesPITR are the same other than the name?

Reply:
NM. tables and tablesPITR are different.

Reply:
Seems not viable after offline discussion.